### ISSUE TITLE
Slower performance after applying the hack Remarkable 2

### ISSUE THREAD

**Issue by dronedad** [ddvk/remarkable-hacks#338]
Just notice my Remarkable 2 was performing slow and I got used to it. The hack benefits are great compared with the standard features that come out of the box. Since I update to the new Remarkable 2 update 2.9.0.153, the device is faster than ever before. My comparison is with the hack installed.
Has anyone experienced slow performance when using the hack? Just wondering what could I do to improve performance after I apply the hack patch again.
Thanks

Andre

**Comment by dronedad** [comment]
I am in Australia. Enrolled to get Beta. So I guess that is why I am running 2.9.

**Comment by pstemper** [comment]
Yes, 2.9 is STILL beta.  Once 2.9 becomes an "official" release, hopefully we will see a DDVK hack for it.  And so far, the DDVK hacks have not appeared to cause a performance issue with either my rM1 or my rM2.

**Comment by dstango** [comment]
is there a way for you to go back to 2.8 and see if the performance difference might be related to the original rm firmware? The rm2 has a more capable processor (dual core I think) than the rm1 but so far it seemed to me they didn't take much advantage of it. Maybe 2.9 brings some improvements in performance because of optimizations taking advantage of the rm2-processor's capabilities.

**Comment by dronedad** [comment]
Thanks for the advice. I will stay on 2.9 for now until some you try it and
share your experiences.

**Comment by dronedad** [comment]
> is there a way for you to go back to 2.8 and see if the performance difference might be related to the original rm firmware? The rm2 has a more capable processor (dual core I think) than the rm1 but so far it seemed to me they didn't take much advantage of it. Maybe 2.9 brings some improvements in performance because of optimizations taking advantage of the rm2-processor's capabilities.

Do you know a way to reverse back to 2.8 - Another feature that is not working is ScreeShare. The Mac desktop app is still 2.8. I want to go back to 2.8 so I can use the LiveView, or similar. I noticed a few hack tool options. What do you recommend?

**Comment by martin-braun** [comment]
Not sure if this was also the case for you, but using custom images like a custom sleeping.png, that are great in size, will eat more RAM, since the tablet seems to keep those images in RAM all the time.

**Comment by dstango** [comment]
You have at least two options:

* simply revert the xochitl binary (the hacks make a backup copy of the original binary): https://github.com/ddvk/remarkable-hacks#revert-in-case-things-go-terribly-wrong -- you might need to then reapply the hacks

* the rm has two "partitions" that can contain different versions of the whole sw setup and you can try to revert to the 2.8-partition. Search the issues for more details how to do that, if you want to pick this option.

I would first try out the first option - it seems to be the easiest way to me and I don't think much can go wrong. (but of course: all on your own risk ;-) ...)

**Comment by dronedad** [comment]
> You have at least two options:
> 
> * simply revert the xochitl binary (the hacks make a backup copy of the original binary): https://github.com/ddvk/remarkable-hacks#revert-in-case-things-go-terribly-wrong -- you might need to then reapply the hacks
> * the rm has two "partitions" that can contain different versions of the whole sw setup and you can try to revert to the 2.8-partition. Search the issues for more details how to do that, if you want to pick this option.
> 
> I would first try out the first option - it seems to be the easiest way to me and I don't think much can go wrong. (but of course: all on your own risk ;-) ...)

Hi. Tried the first option and now I am in trouble. I did try to use the latest I have which is 2.7.0.51. Not 2.8.
Followed the steps and when I after start the device reload but after 'starting' it blink 3 times and then stays with the all white screen. The device is live and I can connect via USB.  Any help? Tried to re-apply the patch which in my case is 2.7.0.51. but always getting the message below:  
The version the device is running is not supported, yet. 20210709092503

Any ideas? Thanks

**Comment by pstemper** [comment]
@dronedad You could try to switch the partition to the "backup" one, as @dstango mentioned.  In issue #304 I give direction step by step on how to do this.  Look in that issue for a posting form me that starts with "No Problem!"
That will switch you to the last partition that was in effect BEFORE you installed the latest rM update..

pstemper

**Comment by dronedad** [comment]
> @dronedad You could try to switch the partition to the "backup" one, as @dstango mentioned. In issue #304 I give directions step by step on how to do this. Look in that issue for a posting form me that starts with "No Problem!"
> That will switch you to the last partition that was in effect BEFORE you installed the latest rM update..
> 
> pstemper

THANKS. It worked :-)

**Comment by pstemper** [comment]
Wonderful!
